src/modules/kvs/kvs.c

```diff
@@ -138,6 +152,16 @@ static struct kvs_ctx *kvs_ctx_create (flux_t *h)
             goto error;
         flux_watcher_start (ctx->prep_w);
         flux_watcher_start (ctx->check_w);
+        /* ctx->kcp will not be created on rank != 0, all kvs_checkpoint
+         * functions will do nothing when ctx->kcp is NULL.
+         */
+        ctx->kcp = kvs_checkpoint_create (h,
+                                          NULL, /* set later */
+                                          0.0,  /* default 0.0, set later */
+                                          work_queue_check_append_wrapper,
+                                          ctx);
+        if (!ctx->kcp)
+            goto error;
```

Comment:
Since this code is inside an `if (ctx->rank == 0)` block, maybe the comment is not needed here?  Might be better to just document in `kvs_checkpoint.h` that each function is a no-op if called with a NULL `kvs_checkpiont_t` argument to avoid excessive rank 0 tests in the code.